### PR TITLE
Axom: Update/merge changes from Axom's repo

### PR DESCRIPTION
**Pull request by white238**
I tested this against Axom's LC use cases

### CODE REVIEW

**Comment by white238** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/axom/package.py [hunk 1]
```diff
@@ -95,11 +97,12 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
     # Dependencies
     # -----------------------------------------------------------------------
     # Basics
-    depends_on("cmake@3.14:", type="build")
+    depends_on("cmake@3.14:", type="build", when="@:0.6.1")
```
Comment (becker33) [inline var/spack/repos/builtin/packages/axom/package.py]:
```suggestion
    depends_on("cmake@3.14:", type="build")
```

File: var/spack/repos/builtin/packages/axom/package.py [hunk 2]
```diff
@@ -231,6 +244,12 @@ def initconfig_compiler_entries(self):
         if "+cpp14" in spec and spec.satisfies("@:0.6.1"):
             entries.append(cmake_cache_string("BLT_CXX_STD", "c++14", ""))
 
+        # Add optimization flag workaround for Debug builds with
+        # cray compiler or newer HIP
+        if "+rocm" in spec:
+            if "crayCC" in self.compiler.cxx or spec.satisfies("%clang@16"):
```
Comment (becker33) [inline var/spack/repos/builtin/packages/axom/package.py]:
```suggestion
            if spec.satisfies("%cce") or spec.satisfies("%clang@16"):
```

File: var/spack/repos/builtin/packages/axom/package.py [hunk 3]
```diff
@@ -266,21 +288,27 @@ def initconfig_hardware_entries(self):
...
-                arch_str = ",".join(archs)
-                entries.append(cmake_cache_string("CMAKE_HIP_ARCHITECTURES", arch_str))
+            # Fix blt_hip getting HIP_CLANG_INCLUDE_PATH-NOTFOUND bad include directory
+            if (
+                self.spec.satisfies("%cce") or self.spec.satisfies("%clang")
+            ) and "toss_4" in self._get_sys_type(spec):
+                # Set the patch version to 0 if not already
+                clang_version = str(self.compiler.version)[:-1] + "0"
+                hip_clang_include_path = (
+                    rocm_root + "/llvm/lib/clang/" + clang_version + "/include"
```
Comment (becker33) [inline var/spack/repos/builtin/packages/axom/package.py:305]: What is going on here? Is this really TOSS specific code?

File: var/spack/repos/builtin/packages/axom/package.py [hunk 4]
```diff
@@ -367,28 +419,40 @@ def initconfig_mpi_entries(self):
         else:
             entries.append(cmake_cache_option("ENABLE_MPI", False))
 
+        # Replace /usr/bin/srun path with srun flux wrapper path on TOSS 4
+        if "toss_4" in self._get_sys_type(spec):
```
Comment (becker33) [inline var/spack/repos/builtin/packages/axom/package.py:425]: Does this mean that axom will be completely broken on any non-LC machine running a scheduler other than slurm?

**Review by becker33 — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/axom/package.py (hunk 4 above)
Comment (white238) [reply var/spack/repos/builtin/packages/axom/package.py:425]: As I understand, this has to do with the default srun of the machine not being the flux wrapped version, which this finds. This is specific to flux machines on LC.

**Comment by white238** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/axom/package.py (hunk 4 above)
Comment (white238) [reply var/spack/repos/builtin/packages/axom/package.py:425]: Logged an issue #42298

**Review by becker33 — APPROVED** [review]

**Pull request merged**